app/models/katello/pool.rb

```diff
@@ -27,9 +41,8 @@ def as_json(*_args)
 
     # If the pool_json is passed in, then candlepin is not hit again to fetch it. This is for the case where
     # prior to this call the pool was already fetched.
-    def self.find_pool(cp_id, pool_json = nil)
-      pool_json = Resources::Candlepin::Pool.find(cp_id) unless pool_json
-      Katello::Pool.new(pool_json) unless pool_json.nil?
+    def self.find_pool(cp_id)
```

Comment:
`by_cp_id` or `with_cp_id` might be a more descriptive name for this function as here it is quite generic and doesn't distinguish itself from the AR 'find'.